cps/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java

```diff
@@ -119,6 +120,18 @@ public String getScriptPath() {
         return exec;
     }
 
+    private FilePath getFilePathWithSuffix(FilePath baseWorkspace) {
+        return baseWorkspace.withSuffix(getFilePathSuffix());
+    }
+
+    private String getFilePathSuffix() {
+        String defined_suffix = new Properties().getProperty("hudson.slaves.WorkspaceList");
```

Comment:
This will not work. It will always be null. Did you test this patch? (`mvn -f aggregator hpi:run -Dhudson.slaves.WorkspaceList=~` or similar)

Best to simply copy the [original definition of `COMBINATOR`](https://github.com/jglick/jenkins/blob/ca3b1db4d6f761861885d03355f35ccafe7cc425/core/src/main/java/hudson/slaves/WorkspaceList.java#L291).